app/controllers/email_subscriptions_controller.rb

```diff
@@ -1,28 +1,25 @@
 class EmailSubscriptionsController < ApplicationController
-  # No need to authorize this because its implicit when unsubbing
+  PREFERRED_EMAIL_NAME = {
+    email_digest_periodic: "DEV digest emails",
+    email_comment_notifications: "comment notifications",
+    email_follower_notifications: "follower notifications",
+    email_mention_notifications: "mention notifications",
+    email_connect_messages: "DEV connect messages",
+    email_unread_notifications: "unread notifications",
+    email_badge_notifications: "badge notifications"
+  }.freeze
+
   def unsubscribe
     verified_params = Rails.application.message_verifier(:unsubscribe).verify(params[:ut])
 
     if verified_params[:expires_at] > Time.current
       user = User.find(verified_params[:user_id])
       user.update(verified_params[:email_type] => false)
-      @email_type = preferred_email_name(verified_params[:email_type])
+      @email_type = PREFERRED_EMAIL_NAME[verified_params[:email_type]]
```

Comment:
Should we maybe use `.fetch` so this fails if the `email_type` is invalid?

Reply:
I ran into this issue already actually. Instead of 
> You've unsubscribed from some email.

you get
> You've unsubscribed from .

Which isn't ideal but isn't breaking anything either. I don't think this needs to be addressed immediately but something to keep in mind.

Reply:
There's always

```ruby
PREFERRED_EMAIL_NAME.fetch(verified_params[:email_type], "this list")
```

if you want to avoid emails like that but don't want to fail either.

Reply:
Actually that's a great idea, I'll go ahead and apply that.